### DISCUSSION TITLE
Allowing reads without users needing to connect a wallet

### DISCUSSION THREAD

**Discussion by kevinhalliday** [wevm/wagmi#70] [Q&A] [answered]
Excited about this library. Loving the hooks/docs/typescript support - super helpful for those of us onboarding to dapp development. Thanks! Wondering what the best flow is allowing read only connections to some alchemy/infura by default, without a user needing to connect their wallet?

**Comment by tmm** [comment, marked as answer]
You can [configure a global ethers.js Provider](https://wagmi-xyz.vercel.app/docs/provider#provider-optional) and access it with the [`useProvider`](https://wagmi-xyz.vercel.app/docs/hooks/useProvider) hook. This works even if an account is not connected.

**Reply by kevinhalliday** [reply]
Oh great - so the answer was in the docs. Looks like I don't fully understand how providers work. Thanks for the quick response

**Reply by isaac-martin** [reply]
Can set up a provider but i'm a bit lost to what to do now? I pass the provider into the `useContractRead` hook but still can't get data unless connected to a wallet 🤔

**Reply by isaac-martin** [reply]
i'm a dummy - without a wallet connected it defaults to mainnet which i don't have a contract deployed to rn. 
for anyone who stumbles across this error while in development you can do this. 

```jsx
 <Provider
        autoConnect
        connectors={connectors}
        // force providers to rinkeby for now
        provider={() => provider({ chainId: 4 })}
        webSocketProvider={() => webSocketProvider({ chainId: 4 })}>
       {children}
      </Provider>
      ```
